### ISSUE TITLE
does age introduce a unique on disk data structure that stores the graphs?

### ISSUE THREAD

**Issue by OpenCoderX** [apache/age#381]
Is there a distinct on disk data structure used to store the graphs or do we transpile cypther to sql then execute the sql against the relational table data structures?

**Comment by imranzaheer612** [comment]
I think it is done using the Postgres relational tables. 
The `ag_graph` and `ag_label` is stored as a relation under `ag_catalog` scheme

```
demo=# \dt
           List of relations
   Schema   |   Name   | Type  | Owner 
------------+----------+-------+-------
 ag_catalog | ag_graph | table | imran
 ag_catalog | ag_label | table | imran
 public     | company  | table | imran
 public     | employee | table | imran
(4 rows)
```

As you can see that my new graph `cypher` is stored as schema. The `id` and `properties` data of the nodes is saved in the table name `_ag_label_vertex`. 

![working](https://user-images.githubusercontent.com/62287456/206905292-8cbc958c-9840-42f7-aa3e-fa5890fe6912.png)

So I think it will be like, first we parse the `cypher` function and then perform corresponding psql operations defined.

**Comment by OpenCoderX** [comment]
Thank you for the explanation, I'm weighing options to replace neo4j. Given that AGE uses typical relational tables, am I to assume that I can expect relational performance when executing deep graph queries?

**Comment by JoshInnis** [comment]
no. AGE has the ability to meet traditional queries where it is necessary and it can surpass relational queries. We need ne details to help you.

**Comment by eyab** [comment]
@OpenCoderX Any updates on this issue?
